File: src/main/java/infinitespire/powers/SuperSlowPower.java
package infinitespire.powers;

import com.megacrit.cardcrawl.actions.common.ApplyPowerAction;
import com.megacrit.cardcrawl.actions.utility.UseCardAction;
import com.megacrit.cardcrawl.cards.AbstractCard;
import com.megacrit.cardcrawl.cards.DamageInfo;
import com.megacrit.cardcrawl.core.CardCrawlGame;
import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
import com.megacrit.cardcrawl.localization.PowerStrings;
import com.megacrit.cardcrawl.monsters.AbstractMonster;
import com.megacrit.cardcrawl.powers.AbstractPower;
import infinitespire.InfiniteSpire;

public class SuperSlowPower extends AbstractPower {

    public static final String powerID = InfiniteSpire.createID("ShreddedPower");
    private static final PowerStrings strings = CardCrawlGame.languagePack.getPowerStrings(powerID);

    public SuperSlowPower(AbstractMonster owner, int amount){
        this.owner = owner;
        this.amount = amount;
        this.name = strings.NAME;
        this.ID =  powerID;
        this.img = InfiniteSpire.Textures.getPowerTexture("superslow.png");
        this.type = PowerType.DEBUFF;
        this.updateDescription();
    }

    @Override
    public void onAfterUseCard(final AbstractCard card, final UseCardAction action) {
        AbstractDungeon.actionManager.addToBottom(new ApplyPowerAction(this.owner, this.owner, new SuperSlowPower((AbstractMonster) this.owner, 1), 1));
    }

    @Override
    public float atDamageReceive(final float damage, final DamageInfo.DamageType type) {
        if (type == DamageInfo.DamageType.NORMAL) {
            return damage * (1.0f + this.amount * 0.1f);
        }
        return damage;
    }

    @Override
    public void updateDescription() {
        this.description =
            strings.DESCRIPTIONS[0]
                + owner.name
                + strings.DESCRIPTIONS[1]
                + amount * 10
                + strings.DESCRIPTIONS[2];
    }
}

File: src/main/kotlin/com/blanktheevil/infinitespire/powers/SuperSlowPower.kt
package com.blanktheevil.infinitespire.powers

import com.blanktheevil.infinitespire.extensions.applyPower
import com.blanktheevil.infinitespire.extensions.makeID
import com.blanktheevil.infinitespire.powers.utils.PowerBuilder
import com.megacrit.cardcrawl.actions.utility.UseCardAction
import com.megacrit.cardcrawl.cards.AbstractCard
import com.megacrit.cardcrawl.cards.DamageInfo
import com.megacrit.cardcrawl.monsters.AbstractMonster
import com.megacrit.cardcrawl.powers.AbstractPower

class SuperSlowPower(
  owner: AbstractMonster,
  amount: Int
) : Power(owner, amount, BUILDER) {
  companion object {
    val powerID = "ShreddedPower".makeID()
    private val BUILDER = PowerBuilder(powerID)
      .debuff()
      .img("superslow.png")
  }

  override fun updateDesc() {
    this.description = StringBuilder(strings.DESCRIPTIONS[0])
      .append(owner.name)
      .append(strings.DESCRIPTIONS[1])
      .append(amount.times(10))
      .append(strings.DESCRIPTIONS[2])
      .toString()
  }

  override fun makeCopy(): AbstractPower = SuperSlowPower(owner as AbstractMonster, amount)

  override fun onAfterUseCard(card: AbstractCard?, action: UseCardAction?) {
    owner.applyPower(
      SuperSlowPower(owner as AbstractMonster, 1)
    )
  }

  override fun atDamageReceive(damage: Float, damageType: DamageInfo.DamageType?): Float {
    return if (damageType != null && damageType == DamageInfo.DamageType.NORMAL) {
      damage * (1.0f + amount * 0.1f)
    } else {
      damage
    }
  }
}